### PR TITLE
Refactor manifest updater

### PR DESCRIPTION
**Pull request by ppoelzl**
### Description of the problem being solved:
Currently, new files have to be added to the manifest manually. This PR automates the inclusion of new files in certain directories, based on the configured include/exclude directives.

### Steps taken to verify a working solution:
Run the script and verify that the output matches the existing script. Note that this only works if no files have changed since the last release. You possibly need to check out a specific commit to test this script.
Might add automatic tests in the future, but this script is hard to mock.

### CODE REVIEW

File: update_manifest.py [hunk 1]
```diff
@@ -1,56 +1,135 @@
...
     file_name = "manifest.xml" if replace else "manifest-updated.xml"
-    manifest.write(base_path / file_name, encoding="UTF-8", xml_declaration=True)
+    tree = Et.ElementTree(root)
+    tree.write(base_path / file_name, encoding="UTF-8", xml_declaration=True)
+    if version is not None:
+        logging.info(f"Updated to version {version}")
 
 
-def cli():
+def cli() -> None:
```
Comment (Wires77) [inline update_manifest.py:132]: Dynamically typed language with return types? 💯

**Review by Wires77 — COMMENTED** [review]
I'm not sure exactly what changed it, but when I run the script through git bash on Windows (how I've done it before, iirc) all of the line breaks get removed.  This is when using `--in-place` if that matters

File: update_manifest.py [hunk 2]
```diff
@@ -1,56 +1,135 @@
...
+    root = Et.Element("PoBVersion")
+    Et.SubElement(root, "Version", number=new_version)
+    for attributes in parts:
+        Et.SubElement(root, "Source", attributes)
+    for attributes in files:
+        Et.SubElement(root, "File", attributes)
     file_name = "manifest.xml" if replace else "manifest-updated.xml"
-    manifest.write(base_path / file_name, encoding="UTF-8", xml_declaration=True)
+    tree = Et.ElementTree(root)
+    tree.write(base_path / file_name, encoding="UTF-8", xml_declaration=True)
```
Comment (Wires77) [inline update_manifest.py:127]: I had to insert `Et.indent(tree, '\t')` here to fix the all-on-one-line issue I mentioned.

File: update_manifest.py [hunk 3]
```diff
@@ -1,56 +1,135 @@
...
-        logger.info(f"Path: {path} hash: {sha1_hash}")
-    if version is not None:
-        root.find("Version").set("number", version)
-        logger.info(f"Updated to version {version}")
+            sha1 = hashlib.sha1(data).hexdigest()
+            name = path.relative_to(config[section]["path"]).as_posix()
+            attributes = {"part": section, "sha1": sha1, "name": name}
+            if path.suffix in [".dll", ".exe"]:
+                attributes.update(runtime="win32")
+            files.append(attributes)
```
Comment (Wires77) [inline update_manifest.py:115]: I like the look of the new manifest, as it lines everything up, but is there a way to order the attributes at least on this first run to verify the diff didn't change?  Right now everything looks like it changed, as the attributes changed order (before was name, part, sha1; now is part, sha1, name)

File: manifest.cfg [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+path =
+include-files = changelog.txt,LICENSE.md
+include-directories =
+
+[runtime]
+path = runtime
+exclude-files = lua-profiler.lua,msvcr100.dll,SimpleGraphic.cfg,Update.exe
+exclude-directories =
+
+[src]
```
Comment (Wires77) [inline manifest.cfg:11]: Old part name was "program", this changes it to "src", possibly messing with the update script (haven't verified yet, but would be good to change it back for the first run at least to verify no changes)

**Review by Wires77 — CHANGES_REQUESTED** [review]
Was able to fix the new line issue, but still found some differences that are inconsequential in the long term, but make it hard to verify unchanging functionality

File: update_manifest.py (hunk 1 above)
Comment (ppoelzl) [reply update_manifest.py:136]: And with decent static type checking as well (if you want to)!

**Review by Wires77 — APPROVED** [review]
This looks good.  I'll have to make sure to clean out my temp files when I run this now (i.e. it's picking up my old `Builds` directory, and some code coverage files at the moment)

I do question the readability of the value -> if clause -> else clause in this new Python syntax, but I'm not a Python expert, so I'll defer to new syntax

**Pull request merged**